### PR TITLE
fix(sdk): Compiler - Fixed pipeline parameters with empty default values

### PR DESCRIPTION
**Pull request by Ark-kun**
Fixes https://github.com/kubeflow/pipelines/issues/4549

### CODE REVIEW

**Comment by Bobgy** [comment]
/assign @numerology @chensun @neuromage 
for review

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -820,23 +817,23 @@ def _create_workflow(self,
     self._sanitize_and_inject_artifact(dsl_pipeline, pipeline_conf)
 
     # Fill in the default values.
-    args_list_with_defaults = []
```
Comment (chensun) [inline sdk/python/kfp/compiler/compiler.py:823]: What is the motivation of changing this from List to OrderedDict? Seems to me we still only iterate through it as if it's a sequential structure. I don't see any usage like key-lookup or similar. Did I miss anything?

**Review by chensun — COMMENTED** [review]
Overall LGTM, only one minor question.

File: sdk/python/kfp/compiler/compiler.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/compiler/compiler.py:823]: `args_list_with_defaults` is passed to `_create_pipeline_workflow` which now expects a dictionary, not a list of `PipelineParam`s. Line 629 iterates through the keys and values of the dictionary.

**Comment by Ark-kun** [comment]
Friendly ping

**Pull request merged**